src/globalPlug.ts

```diff
@@ -0,0 +1,192 @@
+
+            if (typeof controller.enable === 'function') {
+                const promise = controller.enable();
+
+                if (promise instanceof Promise) {
+                    initialization.push(
+                        promise
+                            .then(() => logger.debug(`Plugin "${pluginName}" enabled`))
+                            .catch(() => logger.error(`Failed to enable plugin "${pluginName}"`)),
+                    );
+                }
+            }
+        }
+
+        Promise.all(initialization).then(() => {
+            this.initialize();
+
+            logger.debug('Initialization complete');
+        });
+    }
+
+    public get plugged(): Promise<this> {
+        return this.initialized.then(() => this);
+    }
+
+    public get flushed(): Promise<this> {
+        return this.tracker.flushed.then(() => this);
+    }
+
+    public get sdk(): SdkFacade {
+        if (this.instance === undefined) {
+            throw new Error('Croct is not plugged in.');
+        }
+
+        return this.instance;
+    }
+
+    public get tracker(): TrackerFacade {
+        return this.sdk.tracker;
+    }
+
+    public get user(): UserFacade {
+        return this.sdk.user;
+    }
+
+    public get session(): SessionFacade {
+        return this.sdk.session;
+    }
+
+    public isAnonymous(): boolean {
+        return this.sdk.context.isAnonymous();
+    }
+
+    public getUserId(): string | null {
+        return this.sdk.context.getUser();
+    }
+
+    public identify(userId: string): void {
+        this.sdk.identify(userId);
+    }
+
+    public anonymize(): void {
+        this.sdk.anonymize();
+    }
+
+    public setToken(token: string): void {
+        this.sdk.setToken(token);
+    }
+
+    public unsetToken(): void {
+        this.sdk.unsetToken();
+    }
+
+    public track<T extends ExternalEventType>(type: T, payload: ExternalEventPayload<T>): Promise<ExternalEvent<T>> {
+        return this.sdk.track(type, payload);
+    }
+
+    public evaluate(expression: string, options: EvaluationOptions = {}): Promise<JsonValue> {
+        return this.sdk.evaluate(expression, options);
+    }
+
+    public async unplug(): Promise<void> {
+        if (this.instance === undefined) {
+            return;
+        }
+
+        const logger = this.sdk.getLogger();
+        const pending: Promise<void>[] = [];
+
+        for (const [pluginName, controller] of Object.entries(this.plugins)) {
+            if (typeof controller.disable !== 'function') {
+                continue;
+            }
+
+            logger.debug(`Disabling plugin "${pluginName}"...`);
+
+            const promise = controller.disable();
+
+            if (promise instanceof Promise) {
+                pending.push(
+                    promise
+                        .then(() => logger.debug(`Plugin "${pluginName}" disabled`))
+                        .catch(() => logger.error(`Failed to disable "${pluginName}"`)),
+                );
+            }
```

Comment:
Also here it can should be logged even if its not a Promise

Reply:
Agreed